Sources/AsyncQueue/ActorQueue.swift

```diff
@@ -84,10 +71,30 @@ public final class ActorQueue<ActorType: Actor>: @unchecked Sendable {
     /// **Must be called prior to enqueuing any work on the receiver.**
     ///
     /// - Parameter actor: The actor on which the queue's task will execute. This parameter is not retained by the receiver.
-    /// - Warning: Calling this method more than once will result in an assertion failure.
+    /// - Precondition: Calling this method more than once will result in a precondition failure.
     public func adoptExecutionContext(of actor: ActorType) {
-        assert(weakExecutionContext == nil) // Adopting multiple executionContexts on the same queue is API abuse.
+        precondition(weakExecutionContext == nil) // Adopting multiple executionContexts on the same queue is API abuse.
         weakExecutionContext = actor
+
+        actor.execute { [taskStream] _ in
+            func beginExecuting(
+                _ operation: sending @escaping (isolated ActorType) async -> Void,
+                in context: isolated ActorType
+            ) {
+                // In Swift 6, a `Task` enqueued from an actor begins executing immediately on that actor.
+                // Since we're running on our actor's context due to the isolated parmater, we can just dispatch a Task to get first-enqueued-first-start task execution.
+                Task {
+                    await operation(context)
+                }
+            }
+
+            for await actorTask in taskStream {
+                await beginExecuting(
```

Comment:
we need to `await` here despite the fact that we're already isolated to the `context` because the compiler does not have a guarantee that `actorTask.executionContext === `actor`. In practice, these are always the same and therefore we do not suspend here.